### ISSUE TITLE
Missing clipping of dtype limits in is_low_contrast function

### ISSUE THREAD

**Issue by jo-fra** [scikit-image/scikit-image#6774]
When using the `skimage.exposure.is_low_contrast` function I experience different results after

1. passing a grayscale image of type uint8 with pixel range [0, 255] that I converted beforehand
2. passing an RGB image and let it be converted within `is_low_contrast` by line:
    https://github.com/scikit-image/scikit-image/blob/5e74a4a3a5149a8a14566b81a32bb15499aa3857/skimage/exposure/exposure.py#L833

The reason is that after conversion by `rgb2gray` the image is of type float64 and the pixel range [0.0, 1.0].
However, in the following line `dtype_limits` is called with `clip_negative=False`:
https://github.com/scikit-image/scikit-image/blob/5e74a4a3a5149a8a14566b81a32bb15499aa3857/skimage/exposure/exposure.py#L835
Which results in `dlimits=(-1,1)` and an unexpected value of the calculated ratio.

I expect `dtype_limits` should be called with `clip_negative=True` to receive the expected ratio.

**Comment by stefanv** [comment]
Yeah, yikes, this is a bug, but also a subtle one because technically float dtypes *do* support negative numbers. @jni until skimage 2.0, would it make sense to change that range to (0, 1) everywhere instead of (-1, 1)?
